### ISSUE TITLE
Add cursor trails!

### ISSUE THREAD

**Issue by Sakyce** [VirtCode/hypr-dynamic-cursors#6]
Would be super cool to have that dithering effect like on Hypnospace!!!
https://fxtwitter.com/jaytholen/status/824335885676244993

The trail duration should be configurable, and big bonus if we can use it with other effects.

**Comment by VirtCode** [comment]
haha yes, cursor trails would certainly be fun and nice to have in hyprland

I am however unsure whether they are in scope for this plugin... I have played around a bit with graphically advanced effects (mainly [inverted cursors](https://github.com/VirtCode/hypr-dynamic-cursors/tree/inverted)), and come to the conclusion that they are probably out of scope here as they require heavy modifications to the rendering, and also deviate quite a bit from the original goal of this plugin, adding small transforms to the cursor to make it more "realistic". While a cursor trail wouldn't be that advanced, it would require it's completely own rendering for rendering the "old shapes".

So as it currently stands, I probably won't implement more advanced effects in this plugin, but am considering making a separate plugin which adds more intensive (and costly) cursor effects. I don't have an ETA on that though, and need yet to work out how compatibility between that and this plugin would be. I also have other things to focus on with this plugin, mainly allowing different effect settings per cursor shape (at least with serverside cursors). 

But thanks a lot for the suggestion! I will come back to this issue if I decide to implement it somewhere.

**Comment by DADA30000** [comment]
> haha yes, cursor trails would certainly be fun and nice to have in hyprland
> 
> I am however unsure whether they are in scope for this plugin... I have played around a bit with graphically advanced effects (mainly [inverted cursors](https://github.com/VirtCode/hypr-dynamic-cursors/tree/inverted)), and come to the conclusion that they are probably out of scope here as they require heavy modifications to the rendering, and also deviate quite a bit from the original goal of this plugin, adding small transforms to the cursor to make it more "realistic". While a cursor trail wouldn't be that advanced, it would require it's completely own rendering for rendering the "old shapes".
> 
> So as it currently stands, I probably won't implement more advanced effects in this plugin, but am considering making a separate plugin which adds more intensive (and costly) cursor effects. I don't have an ETA on that though, and need yet to work out how compatibility between that and this plugin would be. I also have other things to focus on with this plugin, mainly allowing different effect settings per cursor shape (at least with serverside cursors).
> 
> But thanks a lot for the suggestion! I will come back to this issue if I decide to implement it somewhere.

Also it would be nice to have cursor trails like in osu, or like in hyprtrails, or just a line that fades generally :)

**Comment by CrossScarDev** [comment]
Am I wrong, but aren't cursor trails in the official hyprwm/hyprland-plugins repo.

**Comment by VirtCode** [comment]
Not to my knowledge, but I might be wrong. If you are referring to [hyprtrails](https://github.com/hyprwm/hyprland-plugins/tree/main/hyprtrails) though, that's only for trails behind windows and not a cursor trail.

**Comment by CrossScarDev** [comment]
Oh, yeah, that's what I was thinking of.

**Comment by DADA30000** [comment]
> Not to my knowledge, but I might be wrong. If you are referring to [hyprtrails](https://github.com/hyprwm/hyprland-plugins/tree/main/hyprtrails) though, that's only for trails behind windows and not a cursor trail.

yes, I know that that's only for trails behind windows, I meant to make cursor trail LIKE a trail behind windows in hyprtrails, just smaller ofc
